### PR TITLE
Replace PushStage-based Filter with GraphStage (part of #19834)

### PR DESCRIPTION
**Pull request by miciek**
Part of #19834

### CODE REVIEW

File: akka-stream-tests/src/test/scala/akka/stream/impl/fusing/InterpreterSupervisionSpec.scala [hunk 1]
```diff
@@ -286,22 +288,6 @@ class InterpreterSupervisionSpec extends AkkaSpec with GraphInterpreterSpecKit {
       }
     }
 
-    "resume when Filter throws" in new OneBoundedSetup[Int](Seq(
```
Comment (miciek) [inline akka-stream-tests/src/test/scala/akka/stream/impl/fusing/InterpreterSupervisionSpec.scala]: Moved to `FlowFilterSpec`.

**Comment by ktoso** [comment]
OK TO TEST

File: akka-stream-tests/src/test/scala/akka/stream/impl/fusing/InterpreterSpec.scala [hunk 2]
```diff
@@ -79,9 +79,9 @@ class InterpreterSpec extends AkkaSpec with GraphInterpreterSpecKit {
       lastEvents() should be(Set(OnComplete))
     }
 
-    "implement one-to-many many-to-one chain correctly" in new OneBoundedSetup[Int](Seq(
```
Comment (miciek) [inline akka-stream-tests/src/test/scala/akka/stream/impl/fusing/InterpreterSpec.scala]: @agolubev What do you think about moving these tests to `FlowFilterSpec`?

**Comment by miciek** [comment]
@ktoso It's not OK TO TEST. You are too fast...

File: akka-stream-tests/src/test/scala/akka/stream/impl/fusing/InterpreterSpec.scala (hunk 2 above)
Comment (agolubev) [inline akka-stream-tests/src/test/scala/akka/stream/impl/fusing/InterpreterSpec.scala]: I would left it here as all graph interpreter test cases in this spec (with checking interpreter events).

**Comment by viktorklang** [comment]
@miciek Wow, a PR where I have nothing to comment on!

**Comment by johanandren** [comment]
LGTM, thanks!

File: akka-stream-tests/src/test/scala/akka/stream/impl/fusing/InterpreterSpec.scala [hunk 3]
```diff
@@ -79,9 +79,9 @@ class InterpreterSpec extends AkkaSpec with GraphInterpreterSpecKit {
...
-      Doubler(),
-      Filter((x: Int) ⇒ x != 0, stoppingDecider))) {
+    "implement one-to-many many-to-one chain correctly" in new OneBoundedSetup[Int](
+      Doubler().toGS,
```
Comment (ktoso) [inline akka-stream-tests/src/test/scala/akka/stream/impl/fusing/InterpreterSpec.scala]: Would be even better to convert those to GraphStages directly so we can move on with dropping PushPullStage :)

Comment (ktoso) [inline akka-stream-tests/src/test/scala/akka/stream/impl/fusing/InterpreterSpec.scala]: I made another ticket for that https://github.com/akka/akka/issues/20288

**Comment by ktoso** [comment]
LGTM, thanks

**Pull request merged**